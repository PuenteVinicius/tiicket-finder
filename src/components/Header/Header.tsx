import { Box } from "grommet";
import styled from "styled-components/macro";

const HeaderTitle = styled.h1`
  display: block;
  width: 100%;
  flex-flow: column;
  text-align: center;
  line-height: 36px;
  color: white;
`;

function Header() {
  return (
    <Box
      direction="row"
      border={{ color: "brand", size: "large" }}
      pad="xlarge"
      align="center"
      alignContent="center"
      alignSelf="center"
      background="brand"
    >
      <HeaderTitle>Ticket Finder</HeaderTitle>
    </Box>
  );
}

export default Header;
