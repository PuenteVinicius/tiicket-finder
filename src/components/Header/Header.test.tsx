import { render } from "@testing-library/react";
import "jest-styled-components";
import Header from "./Header";
describe("Header component", () => {
  let headerElement: any;

  beforeEach(() => {
    headerElement = render(<Header />);
  });

  it("renders home correctly", () => {
    expect(headerElement).toMatchSnapshot();
  });

  it("Should render the tile of the aplication", () => {
    expect(headerElement.getByText("Ticket Finder")).toBeInTheDocument();
  });
});
