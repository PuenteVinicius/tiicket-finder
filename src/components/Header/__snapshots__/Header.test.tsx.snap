// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders home correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  background-color: #7D4CDB;
  color: #f8f8f8;
  border: solid 12px #7D4CDB;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 96px;
}

.c1 {
  display: block;
  width: 100%;
  -webkit-flex-flow: column;
  -ms-flex-flow: column;
  flex-flow: column;
  text-align: center;
  line-height: 36px;
  color: white;
}

@media only screen and (max-width:768px) {
  .c0 {
    border: solid 6px #7D4CDB;
  }
}

@media only screen and (max-width:768px) {
  .c0 {
    padding: 48px;
  }
}

<body>
    <div>
      <div
        class="c0"
      >
        <h1
          class="c1"
        >
          Ticket Finder
        </h1>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="StyledBox-sc-13pk1d4-0 gczVjq"
    >
      <h1
        class="Header__HeaderTitle-oil6jy-0 cZfvjy"
      >
        Ticket Finder
      </h1>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
